src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java

```diff
@@ -157,21 +159,29 @@ public AuthorizationIdentity getAuthorizationIdentity() {
             return entry == null ? AuthorizationIdentity.EMPTY : AuthorizationIdentity.basicIdentity(entry.getAttributes());
         }
 
+        @Override
         public boolean verifyCredential(final Object credential) throws RealmUnavailableException {
-            if (credential instanceof char[]) try {
-                final SimpleRealmEntry entry = map.get(name);
+            try {
+                SimpleRealmEntry entry = map.get(name);
                 if (entry == null) {
                     return false;
                 }
-                final Password password = entry.getPassword();
-                return PasswordFactory.getInstance(password.getAlgorithm()).verify(password, (char[]) credential);
+
+                if (credential instanceof char[]) {
```

Comment:
Should we also expect and accept String as credential ? 

Also, I've noticed that there are ```SecurityRealm``` implementations that only accept ```ClearPassword```. Should we change them as well ?